src/_P096_eInk.ino

```diff
@@ -0,0 +1,603 @@
+              eInkScreen->drawRoundRect(sParams[0].toInt(), sParams[1].toInt(), sParams[2].toInt(), sParams[3].toInt(), sParams[4].toInt(), Plugin_096_ParseColor(sParams[5]));
+            } 
+            else if(subcommand.equalsIgnoreCase(F("px")) && argCount == 3)
+            {
+              eInkScreen->drawPixel(sParams[0].toInt(), sParams[1].toInt(), Plugin_096_ParseColor(sParams[2]));
+            } 
+            else 
+            {
+              success = false;
+            }
+          }
+                    
+        }
+        else
+        {
+          //invalid arguments
+          success = false;
+        }
+
+
+        if(!success)
+        {
+          if (loglevelActiveFor(LOG_LEVEL_INFO)) {
+            addLog(LOG_LEVEL_INFO, F("Fail to parse command correctly; please check API documentation"));
+            String log2  = F("Parsed command = \"");
+            log2 += string;
+            log2 += "\"";
+            addLog(LOG_LEVEL_INFO, log2);
+          }
+        }
+        else
+        {
+          if(plugin_096_sequence_in_progress == false)
+            eInkScreen->display();
+        }
+
+        String log;
+        log.reserve(110);                           // Prevent re-allocation
+        log = F("P096-eInk : WRITE = ");
+        log += tmpString;
+        SendStatus(event->Source, log);             // Reply (echo) to sender. This will print message on browser.  
+              
+        break;        
+      }
+  }
+
+
+
+  return success;
+}
+
+
+//Print some text
+//param [in] string : The text to display
+//param [in] X : The left position (X)
+//param [in] Y : The top position (Y)
+//param [in] textSize : The text size (default 1)
+//param [in] color : The fore color (default ILI9341_WHITE)
+//param [in] bkcolor : The background color (default ILI9341_BLACK)
+void Plugin_096_printText(const char *string, int X, int Y, unsigned int textSize, unsigned short color, unsigned short bkcolor)
+{
+  eInkScreen->clearBuffer();
+  eInkScreen->clearDisplay();
+  eInkScreen->setCursor(X, Y);
+  eInkScreen->setTextColor(color, bkcolor);
+  eInkScreen->setTextSize(textSize);
+  String stringOne  = string;
+  String stringTwo = "";
+  stringTwo += (char)247;
+  stringOne.replace("°", stringTwo);
+  eInkScreen->println(stringOne);
+  eInkScreen->display();
+}
+
+//Parse color string to ILI9341 color
+//param [in] s : The color string (white, red, ...)
+//return : color (default ILI9341_WHITE)
+unsigned short Plugin_096_ParseColor(String & s)
+{
+  if (s.equalsIgnoreCase(F("BLACK")))
+    return EPD_BLACK;
+  if (s.equalsIgnoreCase(F("WHITE")))
+    return EPD_WHITE;
+  if (s.equalsIgnoreCase(F("INVERSE")))
+    return EPD_INVERSE;
+  if (s.equalsIgnoreCase(F("RED")))
+    return EPD_RED;
+  if (s.equalsIgnoreCase(F("DARK")))
+    return EPD_DARK;
+  if (s.equalsIgnoreCase(F("LIGHT")))
+    return EPD_LIGHT;
+  return EPD_WHITE; //fallback value
```

Comment:
These calls do compare a lot with `equalsIgnoreCase`.
Maybe make a local copy of the string, convert it to lower case and compare with `.equals` on that lower case string.
Compares with ignore case are quite resource intensive.

Reply:
ok